scenes/game/local_game.gd

```diff
@@ -3177,22 +3254,22 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
 			else:
 				_append_log_full(Enums.LogType.LogType_Effect, performing_player, "has no cards in discards to boost with.")
 		"boost_then_strike":
-			var allow_gauge = 'allow_gauge' in effect and effect['allow_gauge']
-			var only_gauge = 'only_gauge' in effect and effect['only_gauge']
-			if performing_player.can_boost_something(allow_gauge, only_gauge, effect['limitation']):
-				events += [create_event(Enums.EventType.EventType_ForceStartBoost, performing_player.my_id, 0, "", allow_gauge, only_gauge, effect['limitation'])]
+			var valid_zones = ['hand']
+			if 'valid_zones' in effect:
+				valid_zones = effect['valid_zones']
+			if performing_player.can_boost_something(valid_zones, effect['limitation']):
+				events += [create_event(Enums.EventType.EventType_ForceStartBoost, performing_player.my_id, 0, "", valid_zones, effect['limitation'])]
 				change_game_state(Enums.GameState.GameState_PlayerDecision)
 				decision_info.clear()
 				decision_info.type = Enums.DecisionType.DecisionType_BoostNow
 				decision_info.player = performing_player.my_id
-				decision_info.allow_gauge = allow_gauge
-				decision_info.only_gauge = only_gauge
+				decision_info.valid_zones = valid_zones
 				decision_info.limitation = effect['limitation']
 				performing_player.strike_on_boost_cleanup = true
 				if 'wild_strike' in effect and effect['wild_strike']:
 					performing_player.wild_strike_on_boost_cleanup = true
 			else:
-				if len(performing_player.hand) == 0 or only_gauge: # Avoid leaking information
+				if len(performing_player.hand) == 0 or 'hand' not in valid_zones: # Avoid leaking information
```

Comment:
probably don't need this here either, as currently this is jacko character action and not optional.